### DISCUSSION TITLE
CheckUsersFollowPlaylistRequest is not accurate with current documentation

### DISCUSSION THREAD

**Discussion by Maskowh** [spotify-web-api-java/spotify-web-api-java#395] [General]
Hello,

I think the documentation for [Check if Users Follow Playlist](https://developer.spotify.com/documentation/web-api/reference/check-if-user-follows-playlist) changed, and this lib still use the old format in [CheckUsersFollowPlaylistRequest.java](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/cfd0dae1262bd7f95f90c37b28d27b9c944d471a/src/main/java/se/michaelthelin/spotify/requests/data/follow/CheckUsersFollowPlaylistRequest.java#L109)

What seems to have changed:

The endpoint used in lib is `/v1/users/{owner_id}/playlists/{playlist_id}/followers/contains`
The endpoint in the docutmentation is `/v1/playlists/{playlist_id}/followers/contains`

They seem to have removed the `owner_id` param and no longer mention it in the documentation.

The requests I do with this lib still works, meaning the old endpoint with the `owner_id` param is currently still supported, but it might not be the case in the future.

If you agree this is an issue, I can do a branch to fix, since the fix seems to be easy: I think changing the [endpoint path](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/cfd0dae1262bd7f95f90c37b28d27b9c944d471a/src/main/java/se/michaelthelin/spotify/requests/data/follow/CheckUsersFollowPlaylistRequest.java#L109), removing the [owner_id builder method](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/cfd0dae1262bd7f95f90c37b28d27b9c944d471a/src/main/java/se/michaelthelin/spotify/requests/data/follow/CheckUsersFollowPlaylistRequest.java#L69) and removing the `owner_id` param in the [SpotifyApi method](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/cfd0dae1262bd7f95f90c37b28d27b9c944d471a/src/main/java/se/michaelthelin/spotify/SpotifyApi.java#L780) should be enough.
But I could also try to not introduce breaking changes by only deprecating the [owner_id builder method](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/cfd0dae1262bd7f95f90c37b28d27b9c944d471a/src/main/java/se/michaelthelin/spotify/requests/data/follow/CheckUsersFollowPlaylistRequest.java#L69) instead of removing it for now, and deprecating the method using the `owner_id` param in the [SpotifyApi method](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/cfd0dae1262bd7f95f90c37b28d27b9c944d471a/src/main/java/se/michaelthelin/spotify/SpotifyApi.java#L780), while adding the same method without the `owner_id` param.

It would need to update the [example](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/master/examples/data/follow/CheckUsersFollowPlaylistExample.java) and the [test](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/cfd0dae1262bd7f95f90c37b28d27b9c944d471a/src/test/java/se/michaelthelin/spotify/requests/data/follow/CheckUsersFollowPlaylistRequestTest.java) too.

Thanks for the work on this wrapper and for keeping it uptodate.

**Comment by dargmuesli** [comment]
Hey Maskowh, thank you for the perfect summary! :sparkles: Indeed they seem to have made some changes to the api again and I can't find any update blog post on that. I think your proposal to deprecate the current implementation and add the new is a great way to go :+1: You can go ahead and add this in a PR :)
